applications/RomApplication/python_scripts/petrov_galerkin_training_utility.py

```diff
@@ -0,0 +1,158 @@
+# Import Python modules
+import json
+import numpy as np
+import os
+import scipy.linalg
+from glob import glob
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.RomApplication as KratosROM
+from KratosMultiphysics.RomApplication.randomized_singular_value_decomposition import RandomizedSingularValueDecomposition
+import KratosMultiphysics.kratos_utilities as kratos_utils
+
+def GetFilePath(fileName):
+    return os.path.join(os.getcwd(), fileName)
+
+class PetrovGalerkinTrainingUtility(object):
+    """Auxiliary utility for the Petrov Galerkin training.
+    This class encapsulates all the functions required for the Petrov Galerkin training.
+    These are snapshots collection for the basis Psi used for solving a Petrov Galerkin ROM.
+    The snapshots depends on the basis strategy (i.e. Jacobian or Residuals).
+    """
+
+    def __init__(self, solver, custom_settings):
+        # Validate and assign the HROM training settings
+        settings = custom_settings["train_petrov_galerkin"]
+        settings.ValidateAndAssignDefaults(self.__GetPetrovGalerkinTrainingDefaultSettings())
+
+        # Auxiliary member variables
+        self.solver = solver
+        self.time_step_snapshots_matrix_container = [] ## J@Phi or R
+        self.echo_level = settings["echo_level"].GetInt()
+        self.rom_settings = custom_settings["rom_settings"]
+        self.basis_strategy = settings["basis_strategy"].GetString()
+        self.include_phi = settings["include_phi"].GetBool()
+        self.svd_truncation_tolerance = settings["svd_truncation_tolerance"].GetDouble()
```

Comment:
I think we should discuss about the I/O. I'd expect all these to be within the `rom_settings`.